src/transformStateWithClones.js

```diff
@@ -8,6 +8,37 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const clonedState = [];
+  const currentStateData = { ...state };
+
+  function addDataToArray() {
+    clonedState.push({ ...currentStateData });
+  };
```

Comment:
we don't need to do this, we only need to do it once in each iteration, immediately AFTER switch

Reply:
Thanks, got it!